### PR TITLE
[test] More deserialization recovery vtable layout tests

### PR DESCRIPTION
**Pull request by jrose-apple**
rdar://problem/35508966

### CODE REVIEW

**Comment by jrose-apple** [comment]
@slavapestov Anything else you can think of?

@swift-ci Please test

File: test/Serialization/Recovery/typedefs.swift [hunk 1]
```diff
@@ -144,7 +144,7 @@ open class User {
   // CHECK-RECOVERY: /* placeholder for _ */
   public subscript(_: WrappedInt) -> () { return () }
 
-  // CHECK: subscript<T>(_: T) -> () where T : HasAssoc, T.Assoc == WrappedInt { get }
+  // CHECKx: subscript<T>(_: T) -> () where T : HasAssoc, T.Assoc == WrappedInt { get }
```
Comment (slavapestov) [inline test/Serialization/Recovery/typedefs.swift]: Is CHECKx intentional?

**Review by slavapestov — APPROVED** [review]

File: test/Serialization/Recovery/typedefs.swift (hunk 1 above)
Comment (jrose-apple) [reply test/Serialization/Recovery/typedefs.swift]: Bah, I hadn't rebuilt everything to pick up Nathan's fix for this. Thanks for the catch.

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test and merge

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test and merge

**Pull request merged**